### PR TITLE
Added HVSON8 footprint for NXP PCF8523 RTC

### PR DESCRIPTION
**Pull request by johnfwhitmore**
I've taken a copy of another footprint's yaml definitions as a base to start from. As a result there's a lot of comments as I've not got min and max values. I was hoping to create PR to check with CI but looks like that's not possible.

datasheet:https://www.nxp.com/docs/en/data-sheet/PCF8523.pdf

Screenshot from 2019-04-03 18-32-12

### CODE REVIEW

**Comment by johnfwhitmore** [comment]
There are automatic tests and they worked. But I struggle with sizes of devices, so I've probably not got this 100% as yet. Let me know the required changes and I'll amend.

**Comment by evanshultz** [comment]
You can check out the same tests CI uses at https://github.com/KiCad/kicad-library-utils and run them locally.

Min and max values are given on page 57 of the datasheet. See the table below the package drawing. Enter that info and remove what you don't need instead of leaving comments (naturally comments are OK while we're discussing so I'm taking about in the final entry).

**Comment by johnfwhitmore** [comment]
Scissors missed that page. Thanks I'll get back to you.

**Comment by johnfwhitmore** [comment]
Just remembered why I ignore that page of information. That data on page 57 is for the physical device, I'm trying to enter the data for a footprint. So the pin length on the device has min of 0.4 and a max of 0.65 but the footprint only has a value of 0.95 which isn't between the min and the max.

Similarly the pad under the device should I use the data from the device or the data from the footprint. I've gone with footprint, as that's what I'm creating.

**Comment by johnfwhitmore** [comment]
Screenshot from 2019-04-03 19-01-45

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/hvson8.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+HVSON-8-1EP_3.25x2.35mm_P0.8mm_E1.1x1.75mm:
+  device_type: 'HVSON'
+  library: Package_SON
+  #manufacturer: 'NXP'
+  size_source: 'https://www.nxp.com/docs/en/data-sheet/PCF8523.pdf (page 66)'
+  ipc_class: 'qfn_pull_back' #'qfn' | 'qfn_pull_back'
```
Comment (evanshultz) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/hvson8.yaml]: Use `qfn` as this package doesn't have pull-back leads.

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/hvson8.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+HVSON-8-1EP_3.25x2.35mm_P0.8mm_E1.1x1.75mm:
+  device_type: 'HVSON'
+  library: Package_SON
+  #manufacturer: 'NXP'
+  size_source: 'https://www.nxp.com/docs/en/data-sheet/PCF8523.pdf (page 66)'
```
Comment (evanshultz) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/hvson8.yaml]: Page 57

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/hvson8.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
+HVSON-8-1EP_3.25x2.35mm_P0.8mm_E1.1x1.75mm:
```
Comment (evanshultz) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/hvson8.yaml]: Dimensions are 4x4mm with an EP of 2.2x3.1mm.

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/hvson8.yaml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  device_type: 'HVSON'
+  library: Package_SON
+  #manufacturer: 'NXP'
+  size_source: 'https://www.nxp.com/docs/en/data-sheet/PCF8523.pdf (page 66)'
+  ipc_class: 'qfn_pull_back' #'qfn' | 'qfn_pull_back'
+  #ipc_density: 'least' #overwrite global value for this device.
+  #body_size_x_min: 2.85
+  body_size_x:
+    minimum: 3.9
+    nominal: 4.0
```
Comment (evanshultz) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/hvson8.yaml]: Omit nominal if not given (the scripts calculates all info needed or lets you know what's missing).

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/hvson8.yaml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    maximum: 4.1
+  body_size_y:
+    minimum: 3.9
+    nominal: 4.0
+    maximum: 4.1
+  # overall_height:
+    minimum: 1.00
+  #  nominal: 1.0
```
Comment (evanshultz) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/hvson8.yaml]: Remove these commented lines. Only max is OK for height.

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/hvson8.yaml [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  #  maximum: 0.80
+  lead_width:
+    minimum: 0.30
+    nominal: 0.35
```
Comment (evanshultz) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/hvson8.yaml]: Remove nominal.

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/hvson8.yaml [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    maximum: 0.40
+  lead_len:
```
Comment (evanshultz) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/hvson8.yaml]: Use the min and max values of L and remove the nominal one.

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/hvson8.yaml [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+  #  minimum: 0.4
+    nominal: 0.95
+  #  maximum: 0.65
+
+  EP_size_x:
```
Comment (evanshultz) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/hvson8.yaml]: Use Eh and Dh values for the EP size, min and max with no nominal.

**Comment by evanshultz** [comment]
The scripts generate a footprint from the package dimensions. So you enter the package info.

This is documented in IPC 7351 so search for that document if you want to know the methodology used to derive the footprint from the package design.

I've made some comments inline. While the scripts are powerful the documentation is missing so there's understandably a learning curve.

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/hvson8.yaml (hunk 2 above)
Comment (johnfwhitmore) [reply scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/hvson8.yaml]: ok   :(

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/hvson8.yaml (hunk 3 above)
Comment (johnfwhitmore) [reply scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/hvson8.yaml]: Yes sorry

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/hvson8.yaml (hunk 2 above)
Comment (johnfwhitmore) [reply scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/hvson8.yaml]: OK  IPC 7351 right so there is a reason for this. I'll search for that document, thanks

**Comment by johnfwhitmore** [comment]
#ipc_density: 'least' #overwrite global value for this device.

and

  # EP_paste_coverage: 0.65
  EP_num_paste_pads: [2, 2]

I've only two comments in the file now. I'll look into them in IPC7351

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/hvson8.yaml [hunk 9]
```diff
@@ -0,0 +1,35 @@
+HVSON-8-1EP_4x4mm_P0.8mm_E2.2x3.1mm:
```
Comment (evanshultz) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/hvson8.yaml]: Missing the `P` in `EP`.

Comment (johnfwhitmore) [reply scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/hvson8.yaml]: Oops

**Comment by evanshultz** [comment]
Those comments are fine to leave. They will need a thorough cleaning up later but there's too many of them to worry about individually at this time. Breakdown:
- We're using density B (nominal density) for all footprints now.
- Paste coverage is not in the IPC doc. It's a colloquial choice and the script uses a reasonable default value unless overridden.

**Comment by johnfwhitmore** [comment]
OK thanks for that. I uncommented out the manufacturer. The nxp prefix doesn't look right to me. But I guess it's not a standard device and is NXP specific.

Screenshot from 2019-04-03 19-42-20

**Comment by evanshultz** [comment]
It is standard. Page 57 states this package conforms to JEDEC MO-229.

**Comment by johnfwhitmore** [comment]
So should I remove the manufacturer element?

**Comment by evanshultz** [comment]
Yes please.

**Comment by evanshultz** [comment]
Looks good. Thanks!

**Comment by evanshultz** [comment]
@pointhi 
The footprint has been merged.

**Comment by johnfwhitmore** [comment]
Thank you. I'll try the third device tomorrow and make it a cleaner addition

**Pull request merged**